### PR TITLE
small improvements to the default report

### PR DESCRIPTION
**Pull request by wideglide**
- Placing all coverage growth graphs in the same tabbed section.
- Updating functions to support `bugs_covered` as a metric.
- Adding a boxplot to replace the violin plot which shows
  greater detail, especially when outliers are present.
- There appears to still be a problem with the bug coverage
  growth plot as the lines are not monotonically increasing?
- Fixed drop duplicates as the first column (id) was causing duplicates not to be dropped.  This may not be an issue when querying data from the database.

### CODE REVIEW

File: analysis/plotting.py [hunk 1]
```diff
@@ -248,21 +256,69 @@ def write_violin_plot(self, benchmark_snapshot_df, image_path):
         self._write_plot_to_image(self.violin_plot, benchmark_snapshot_df,
                                   image_path)
 
-    def distribution_plot(self, benchmark_snapshot_df, axes=None):
+    def box_plot(self, benchmark_snapshot_df, axes=None, bugs=False):
```
Comment (lszekeres) [inline analysis/plotting.py:259]: Would it make sense to `merge violin_plot()` and `box_plot()` into a single `box_or_violin_plot(violin=False)` which would draw violin/box depending on the parameter to remove some redundancy?

File: analysis/plotting.py [hunk 2]
```diff
@@ -130,6 +130,13 @@ def _write_plot_to_image(self,
         finally:
             plt.close(fig)
 
+    def _common_datafame_checks(self, benchmark_df, snapshot=False):
+        """assertions common to several plotting functions"""
```
Comment (lszekeres) [inline analysis/plotting.py:134]: nit: """Assertions ... functions." (capital A, period)

File: analysis/plotting.py [hunk 3]
```diff
@@ -174,7 +181,7 @@ def coverage_growth_plot(self,
                     loc='upper left',
                     frameon=False)
 
-        axes.set(ylabel='Bugs coverage' if bugs else 'Edge coverage')
+        axes.set(ylabel='Bug coverage' if bugs else 'Region coverage')
```
Comment (lszekeres) [inline analysis/plotting.py:184]: nit: 'Code coverage' or 'Code region coverage' might be clearer

**Review by lszekeres — COMMENTED** [review]
Thanks! LGTM, left few minor questions

File: analysis/plotting.py (hunk 3 above)
Comment (inferno-chromium) [reply analysis/plotting.py:184]: +1

File: analysis/plotting.py (hunk 2 above)
Comment (inferno-chromium) [reply analysis/plotting.py:134]: +1

File: analysis/plotting.py (hunk 1 above)
Comment (inferno-chromium) [reply analysis/plotting.py:259]: +1

File: analysis/report_templates/default.html [hunk 4]
```diff
@@ -168,46 +168,36 @@ <h5 class="center-align">By avg. rank</h5>
...
                     <h5 class="center-align">Reached code coverage distribution</h5>
                     <img class="responsive-img materialboxed"
-                         src="{{ benchmark.violin_plot }}">
+                         src="{{ benchmark.box_plot }}">
                 </div>
             </div>
             <div class="row">
                 <div class="col s12">
                     <ul class="tabs">
+            {% if benchmark.type == 'bug' %}
```
Comment (inferno-chromium) [inline analysis/report_templates/default.html:192]: If it is bug based coverage, put them in the order first, so they should by default first in first tab.

**Review by inferno-chromium — APPROVED** [review]
LGTM with some changed needed as per Laszlo comments. Thanks a lot for this.

File: analysis/report_templates/default.html (hunk 4 above)
Comment (wideglide) [reply analysis/report_templates/default.html:192]: bug coverage are the first two tabs now.

File: analysis/data_utils.py [hunk 5]
```diff
@@ -44,13 +44,15 @@ def validate_data(experiment_df):
...
-    # Remove duplicate rows where only the |crash_key| column changes.
-    experiment_df = experiment_df.drop(columns='crash_key').drop_duplicates()
-
-    return experiment_df[[
+    columns_to_keep = [
         'benchmark', 'fuzzer', 'trial_id', 'time', 'edges_covered',
         'bugs_covered', 'experiment', 'experiment_filestore'
-    ]]
+    ]
+    # Remove extra columns
```
[the remark this answers is not in the document]
Comment (wideglide) [reply analysis/data_utils.py]: fixed.

File: analysis/plotting.py (hunk 3 above)
Comment (wideglide) [reply analysis/plotting.py]: fixed.

File: analysis/plotting.py (hunk 2 above)
Comment (wideglide) [reply analysis/plotting.py]: fixed.

File: analysis/plotting.py (hunk 1 above)
Comment (wideglide) [reply analysis/plotting.py]: combined!

**Review by lszekeres — APPROVED** [review]
Thanks!

**Review by jonathanmetzman — COMMENTED** [review]
LGTM

**Pull request merged**